internal/pkg/template/override/rule.go

```diff
@@ -0,0 +1,34 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package override
+
+type cfnOverrideRule struct {
+	// ruleString example: "ContainerDefinitions[0].Ulimits.HardLimit: 1024"
```

Comment:
Is that the user experience that we want? like folks writing:
```yaml
taskdef_override:
- "ContainerDefinitions[0].Ulimits.HardLimit: 1024"
```
My understanding was it would be:
```yaml
taskdef_override:
- "ContainerDefinitions[0].Ulimits.HardLimit": 1024
```

If so what do you think of having this:
```go
type Rule struct {
   pathSegment   string
   value         yaml.Marshaler
}

func ParseRule(segment string, value interface{}) (Rule, error) { ... }

// Apply updates the path segment of n with the provided value.
func (r Rule) Apply(n *yaml.Node) error

```

Reply:
The user experience will be
``` yaml
taskdef_override:
  - ContainerDefinitions[0].Ulimits.HardLimit: 1024
```
my plan was to add
``` go
type cfnOverrideRule struct {
	// ruleString example: "ContainerDefinitions[0].Ulimits.HardLimit: 1024"
	ruleString string
        ruleValue interface{}
	ruleNodes  []ruleNode
}

type ruleNode struct {
	name        string
	valueType   string
	listValue   listValue
	mapValue    mapValue
	singleValue interface{}
}

type listValue struct {
	index int
}

type mapValue struct {
	key string
}
```
in the next PR and handle the parsing. Then when applying each rule, we just loop through the node to find the location where we want to apply the override value.

Reply:
Anyway, updated the PR hopefully it would make more sense! Let me know you thought about it